packages/abi-gen/src/utils.ts

```diff
@@ -24,25 +30,51 @@ export const utils = {
                 { regex: '^bytes\\d*$', tsType: 'string' },
             ];
             if (paramKind === ParamKind.Input) {
-                // web3 allows to pass those an non-bignumbers and that's nice
-                // but it always returns stuff as BigNumbers
+                // web3 and ethers allow to pass those as numbers instead of bignumbers
                 solTypeRegexToTsType.unshift({
                     regex: '^u?int(8|16|32)?$',
                     tsType: 'number|BigNumber',
                 });
             }
+            if (backend === ContractsBackend.Ethers) {
+                if (paramKind === ParamKind.Output) {
+                    // ethers-contracts automatically converts small BigNumbers to numbers
+                    solTypeRegexToTsType.unshift({
+                        regex: '^u?int(8|16|32|48)?$',
+                        tsType: 'number',
+                    });
```

Comment:
Maybe I'm not reading this correctly but there is overlap between this conditional and the previous one. Is that so? Ideally there shouldn't be.

Reply:
I don't see the overlap. The parameter kind can't be input and output at the same time